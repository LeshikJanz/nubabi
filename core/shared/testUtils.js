/* eslint-disable import/no-extraneous-dependencies */
import configureMockStore from 'redux-mock-store';
import { createEpicMiddleware } from 'redux-observable';
import firebaseMock from 'firebase-mock';

export const setupEpic = epic => {
  const epicMiddleware = createEpicMiddleware(epic);
  const mockStore = configureMockStore([epicMiddleware]);

  return {
    middleware: epicMiddleware,
    mockStore,
    afterEach: () => {
      epicMiddleware.replaceEpic(epic);
    },
  };
};

export const mockFirebase = () => {
  const mockDb = new firebaseMock.MockFirebase();
  const mockAuth = new firebaseMock.MockFirebase();
  const mockSdk = firebaseMock.MockFirebaseSdk(
    path => {
      return path ? mockDb.child(path) : mockDb;
    },
    () => {
      return mockAuth;
    },
  );

  return mockSdk;
};
