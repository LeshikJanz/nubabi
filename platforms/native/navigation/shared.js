import React from 'react';
import { pathOr } from 'ramda';
import LeftHeader from './LeftHeader';
import RightHeader from './RightHeader';

const babyNameTitle = ({ state }) =>
  pathOr('Nubabi', ['params', 'babyName'], state);

export function getBabyTitle(navigation) {
  if (!navigation) {
    throw new Error('an screen hasnt been updated, check stack');
  }

  return babyNameTitle(navigation);
}

export const getTabHeaders = navigate => ({
  left: <LeftHeader navigate={navigate} />,
  right: <RightHeader navigate={navigate} />,
});
