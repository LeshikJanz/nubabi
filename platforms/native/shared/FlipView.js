// @flow
import React, { Component } from 'react';
import { View, Animated } from 'react-native';

const styles = {
  animatedContainer: {
    flex: 1,
  },
};

type Props = {
  renderFront: Function,
  renderBack: Function,
  velocity: number,
  tension: number,
  friction: number,
};

export class FlipCard extends Component {
  props: Props;

  state = {
    animatedValue: new Animated.Value(0),
    isFlipped: true,
  };

  componentDidUpdate(prevProps: Props, prevState: *) {
    if (this.state.isFlipped !== prevState.isFlipped) {
      this._flippedCard();
    }
  }

  _flipToggleCard = () => {
    this.setState({ isFlipped: !this.state.isFlipped });
  };

  _flippedCard() {
    Animated.spring(this.state.animatedValue, {
      toValue: 0, // Returns to the start
      velocity: this.props.velocity, // Velocity makes it move
      tension: this.props.tension, // Slow
      friction: this.props.friction, // Oscillate a lot
    }).start();
  }

  flippedCardView(isFlipped) {
    if (isFlipped) {
      return this.props.renderFront(this._flipToggleCard);
    }
    return this.props.renderBack(this._flipToggleCard);
  }

  render() {
    const perspective = 2000;
    const rotateY = this.state.animatedValue.interpolate({
      inputRange: [0, 1],
      outputRange: ['0deg', '180deg'],
    });
    return (
      <View style={styles.animatedContainer}>
        <Animated.View
          style={[
            styles.animatedContainer,
            { transform: [{ rotateY }, { perspective }] },
          ]}
        >
          {this.flippedCardView(this.state.isFlipped)}
        </Animated.View>
      </View>
    );
  }
}

export default FlipCard;
