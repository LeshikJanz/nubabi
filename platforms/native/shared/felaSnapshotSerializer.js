// WIP
// prettier-ignore
const felaSnapshotSerializer = {
  test(val) {
    return val && val.felaRules && val.$$typeof === Symbol.for('react.test.json');
  },

  print(val, print) {
    const styles = val.felaRules;
    val.felaRules = false;

    return `style = Object ${JSON.stringify(styles, null, 4)}\n\n${print(val)}`;
  },
};

module.exports = felaSnapshotSerializer;
