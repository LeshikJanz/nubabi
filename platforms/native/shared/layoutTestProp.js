export default {
  fontScale: 2,
  height: 1334,
  scale: 2,
  width: 750,
  viewportWidth: 750,
  viewportHeight: 1334,
};
