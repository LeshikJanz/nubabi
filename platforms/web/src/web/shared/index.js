import * as actions from "./actions";
export { default as containers } from "./containers";
export { default as reducer } from "./reducer";
export { default as epics } from "./epics";
export { actions };
