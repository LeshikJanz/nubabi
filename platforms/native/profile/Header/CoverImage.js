// @flow
import type { LayoutProps, Image as ImageType } from 'core/types';
import React from 'react';
import { View, StyleSheet, Dimensions } from 'react-native';
import { compose } from 'ramda';
import { CachedImage as Image } from 'react-native-cached-image';
import { dontUpdateForUploadedImage, withLayout } from '../../components';

const headerBackground = require('core/images/profileBackground.jpg');

type Props = {
  coverImage: ?ImageType,
  layout: LayoutProps,
};

const CoverImage = ({ coverImage, layout }: Props) => {
  let imageSource;
  if (coverImage) {
    imageSource = { uri: coverImage.url };
  } else {
    imageSource = headerBackground;
  }

  const width = layout ? layout.viewportWidth : Dimensions.get('window').width;

  return (
    <View>
      <Image
        source={imageSource}
        style={{
          width,
          position: 'absolute',
          height: 210,
        }}
      />
      <View style={[styles.imageOverlay, { width }]} />
    </View>
  );
};

const styles = StyleSheet.create({
  imageOverlay: {
    height: 210,
    top: 0,
    left: 0,
    right: 0,
    backgroundColor: '#748294',
    opacity: 0.7,
    position: 'absolute',
    flex: 1,
  },
});

export default compose(withLayout, dontUpdateForUploadedImage('coverImage'))(
  CoverImage,
);
