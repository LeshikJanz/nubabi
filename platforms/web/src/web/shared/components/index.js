import About from "./About";
import AsyncComponent from "./AsyncComponent";
import AuthenticatedRoute from "./AuthenticatedRoute";
import Header from "./Header";
import Home from "./Home";
import Loader from "./Loader";
import Test from "./Test";

export default {
  About,
  AsyncComponent,
  AuthenticatedRoute,
  Header,
  Home,
  Loader,
  Test
};
