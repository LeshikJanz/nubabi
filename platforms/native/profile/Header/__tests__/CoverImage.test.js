import 'react-native';
import React from 'react';
import CoverImage from '../CoverImage';
import { expectRender } from '../../../shared/testUtils';

test('it renders correctly', () => {
  expectRender(<CoverImage />);
});
