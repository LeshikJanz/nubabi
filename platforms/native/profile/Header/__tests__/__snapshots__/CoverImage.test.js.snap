// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
style = Object {}

<View>
  <Image
    source={
      Object {
        "testUri": "../../../core/images/profileBackground.jpg",
      }
    }
    style={
      Object {
        "height": 210,
        "position": "absolute",
        "width": 750,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#748294",
          "flex": 1,
          "height": 210,
          "left": 0,
          "opacity": 0.7,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "width": 750,
        },
      ]
    }
  />
</View>
`;
